storage/series/series.go

```diff
@@ -443,41 +450,56 @@ func (s *dbSeries) Bootstrap(blocks block.DatabaseSeriesBlocks) error {
 	if s.bs == bootstrapping {
 		return errSeriesIsBootstrapping
 	}
+	if bootstrappedBlocks == nil {
+		s.bs = bootstrapped
```

Comment:
Hm, maybe below it's not worth setting `s.bs = bootstrapping` below?  What's going to happen if we never can bootstrap the series?  We at the beginning we should just: 
```go
s.Lock()
defer func() { 
  s.Unlock()
  s.bs = bootstrapped
}()

if s.bs != bootstrapNotStarted {
  return errSeriesAlreadyBootstrapped
}
```

That way it's just a binary not bootstrapped / bootstrapped.  (we can removing the `bootstrapping` type, it's local to this package I believe?).

i.e. even if we hit an error it's probably better to set ourselves as bootstrapped (partially) and let the caller deal with the error as they see fit.

Reply:
I like it, done